python/torch_mlir_e2e_test/test_suite/backprop.py

```diff
@@ -66,9 +66,9 @@ def __init__(self):
     @export
     @annotate_args([
         None,
-        ([-1, -1, -1, -1], torch.float32, True),
-        ([-1, -1, -1, -1], torch.float32, True),
-        ([-1, -1, -1, -1], torch.float32, True),
```

Comment:
Why are these changed to static?

Reply:
Because the updated decomposition now involves `aten.view` op https://github.com/llvm/torch-mlir/blob/3c954a9137cecf253ed7080b4aad3d6af94a8397/lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp#L1580-L1581, and the lowering for view op doesn't support dynamic shapes being passed as an operand.

Reply:
This is a step back for this op. In general, changes should not remove existing support for features that are e2e tested. Is there no way of supporting non-unit padding without removing support for dynamic sizes?

Reply:
TBH I don't see any other way. Because this is how the convolution backwards is implemented in PyTorch. Also, we have a trade-off here, either support this op for all args being non-unit or extend it for all but with static shapes.

Reply:
We should fix the view op to support dynamic shapes. I think that might be a bit of a yak shave but this is probably the final straw and we need to buckle down and do it. Regressing something as basic as dynamic shapes for convolution due to a missing feature in the view lowering seems like a good signal we need to dig in and fix the view op lowering.

Reply:
@silvasean The `tensor::CollapsShape` and `tensor::ExpandShape` op, do not work with the dynamic shapes. That's why we have this restriction.

Reply:
We can fix that or collaborate with the folks that can.

Reply:
Can you fix the upstream tensor dialect ops? Or who would be the appropriate person to fix this?

Reply:
I think at the moment Ramiro would be the best to scope this and make a judgment call about how to move forward.

Reply:
Hi @ramiro050, what are your views on Sean's comments?

Reply:
Hi, sorry I've been a bit busy. I will have an answer for you by Friday

Reply:
I ran the e2e test locally with dynamic sizes and the error I get does not seem to be related to `ViewOp`. I get an error when the `ConvolutionOp` is being converted to linalg because the `groups` argument is not a constant int anymore:

https://github.com/llvm/torch-mlir/blob/9f927a7aa17ba8af79a4c9f054bd87914af35a64/lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp#L1610-L1615

Do you see a similar thing?

Reply:
I don't think that is an issue with `view`. The problem is that in the decomposition, the code that creates `gradWeightViewShapeValue` does not handle dynamic sizes correctly, so the `view` op gets a list of `-1`s for the output size argument.

https://github.com/llvm/torch-mlir/blob/9f927a7aa17ba8af79a4c9f054bd87914af35a64/lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp#L1654-L1656

Reply:
That's true because the code was supposed to handle static cases. Even after fixing that it throws the following error:
```
** Failure : found ambiguous expand of dynamic sizes (e.g. [-1, -1] -> [-1, -1, -1])
```
which is related to view op.

Reply:
I have updated the patch with the changes, you can try the test `ConvolutionBackwardModule2D_basic`, and it will fail with this error. And particularly, this is the case that I initially said is not supported in Torch-MLIR, as well as by the `expand`/`collapse` op of the tensor dialect. Where the expansion of one or more dynamic dims is done to another set of dynamic dims.

Reply:
Thanks, I'll take a look

Reply:
Quick update: There is already some work being done to add dynamic shape support to `tensor.expand_shape` (https://discourse.llvm.org/t/rfc-add-explicit-shape-inputs-to-tensor-expand-shape/65952/16). I will be helping out to get the patch landed. This should fix the dynamic shape limitations.

Reply:
@ramiro050 any update for this one?

Reply:
The patch for `expand_shape` is almost complete. I made an update to it today. All that is missing is review + updating lit tests. Patch: https://reviews.llvm.org/D140821